cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaClusterRebalanceAssemblyOperator.java

```diff
@@ -0,0 +1,546 @@
+                                                log.info("{}: No annotation {}", reconciliation, ANNO_STRIMZI_IO_REBALANCE);
+                                                return Future.succeededFuture();
+                                            }
+                                        }).mapEmpty();
+                            }, exception -> {
+                                    log.error("{}: Status updated to [NotReady] due to error", exception);
+                                    return updateStatus(clusterRebalance, new KafkaClusterRebalanceStatus(), clusterRebalanceOperator, State.NotReady.toString(), exception)
+                                            .mapEmpty();
+                                });
+                });
+    }
+
+    private Future<KafkaClusterRebalanceStatus> onNew(CruiseControlApi apiClient,
+                                                      RebalanceOptions.RebalanceOptionsBuilder rebalanceOptionsBuilder) {
+        return apiClient.rebalance("host", 0, rebalanceOptionsBuilder.build())
+                .map(response -> {
+                    if (response.getJson() != null) {
+                        return new KafkaClusterRebalanceStatusBuilder()
+                                .withOptimizationResult(response.getJson().getJsonObject("rebalance").getMap())
+                                .addNewCondition().withNewType(State.ProposalReady.toString()).endCondition().build();
+                    } else {
+                        return new KafkaClusterRebalanceStatusBuilder()
+                                .withNewSessionId(response.getUserTaskId())
+                                .addNewCondition().withNewType(State.PendingProposal.toString()).endCondition().build();
+                    }
+                });
+    }
+
+    private Future<KafkaClusterRebalanceStatus> onPendingProposal(CruiseControlApi apiClient,
+                                                                  KafkaClusterRebalance clusterRebalance,
+                                                                  RebalanceAnnotation rebalanceAnnotation) {
+        Promise<KafkaClusterRebalanceStatus> p = Promise.promise();
+        if (rebalanceAnnotation == RebalanceAnnotation.none) {
+            log.debug("Arming Cruise Control rebalance user task status timer");
+            String sessionId = clusterRebalance.getStatus().getSessionId();
+            vertx.setPeriodic(REBALANCE_TASK_STATUS_TIMER_MS, t -> {
+                resourceOperator.getAsync(clusterRebalance.getMetadata().getNamespace(), clusterRebalance.getMetadata().getName()).setHandler(getResult -> {
+                    if (getResult.succeeded()) {
+                        KafkaClusterRebalance fetchedClusterRebalance = getResult.result();
+                        if (getRebalanceAnnotation(fetchedClusterRebalance) == RebalanceAnnotation.stop) {
+                            log.debug("Stopping current Cruise Control rebalance user task");
```

Comment:
Log the `Reconciliation`

Reply:
I was avoiding to pass the `Reconciliation` object inside all "onXXXXX" methods trying to log useful information in the calling code.